src/main/java/com/marklogic/kafka/connect/source/MarkLogicSourceConfig.java

```diff
@@ -0,0 +1,87 @@
+package com.marklogic.kafka.connect.source;
+
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigDef.Importance;
+import org.apache.kafka.common.config.ConfigDef.Type;
+
+import java.util.Map;
+
+/**
+ * Defines configuration properties for the MarkLogic source connector.
+ */
+public class MarkLogicSourceConfig extends AbstractConfig {
+
+    public static final String CONNECTION_HOST = "ml.connection.host";
+    public static final String CONNECTION_PORT = "ml.connection.port";
+    public static final String CONNECTION_DATABASE = "ml.connection.database";
+    public static final String CONNECTION_SECURITY_CONTEXT_TYPE = "ml.connection.securityContextType";
+    public static final String CONNECTION_USERNAME = "ml.connection.username";
+    public static final String CONNECTION_PASSWORD = "ml.connection.password";
+    public static final String CONNECTION_TYPE = "ml.connection.type";
+    public static final String CONNECTION_SIMPLE_SSL = "ml.connection.simpleSsl";
+    public static final String CONNECTION_CERT_FILE = "ml.connection.certFile";
+    public static final String CONNECTION_CERT_PASSWORD = "ml.connection.certPassword";
+    public static final String CONNECTION_EXTERNAL_NAME = "ml.connection.externalName";
+    public static final String ENABLE_CUSTOM_SSL = "ml.connection.enableCustomSsl";
+    public static final String TLS_VERSION = "ml.connection.customSsl.tlsVersion";
+    public static final String SSL_HOST_VERIFIER = "ml.connection.customSsl.hostNameVerifier";
+    public static final String SSL_MUTUAL_AUTH = "ml.connection.customSsl.mutualAuth";
+
+    public static final String DMSDK_BATCH_SIZE = "ml.dmsdk.batchSize";
+    public static final String DMSDK_THREAD_COUNT = "ml.dmsdk.threadCount";
+
+    public static final String BULK_DS_ENDPOINT_URI = "ml.source.bulkds.endpointUri";
```

Comment:
Don't need the bulk stuff (hopefully we never do, as based on the DHF Spark 2 code, it's pretty difficult to custom code this stuff)